kyo-data/shared/src/main/scala/kyo/Span.scala

```diff
@@ -137,18 +140,182 @@ object Span:
           */
         inline def apply(idx: Int): A = self(idx)
 
+        /** Selects the first element.
+          *
+          * @return
+          *   the first element if the Span is not empty, Absent otherwise
+          */
+        inline def head: Maybe[A] =
```

Comment:
The APIs are also designed to be safer than the counterparts in stdlib

Reply:
`Maybe` unfortunately will box for primitive values due to the fact that it is a union type under the hood but at least there isn't an extra wrapper object like with `Option`